parser_library/src/processing/preprocessors/cics_preprocessor.cpp

```diff
@@ -1114,6 +1193,44 @@ class cics_preprocessor final : public preprocessor
         return events;
     }
 
+    bool try_dfh_lookup(preprocessor::line_iterator& it, const preprocessor::line_iterator& end, const auto lineno)
+    {
+        static const std::regex dfh_lookup("([^ ]*)[ ]+([A-Z#$@][A-Z#$@0-9]*)[ ]+((?:\\S+,)?(?:DFHRESP|DFHVALUE)[ "
+                                           "]*\\([ ]*[A-Z0-9]*[ ]*\\))(?:[ ]+(.*))?",
+            std::regex_constants::icase);
+
+        bool ret_val = false;
+        it = extract_nonempty_logical_line(m_logical_line, it, end, lexing::default_ictl);
+
+        if (m_logical_line.continuation_error)
+        {
+            if (m_diags)
+                m_diags->add_diagnostic(diagnostic_op::warn_CIC001(range(position(lineno, 0))));
+        }
+        else if (std::regex_match(m_logical_line.begin(), m_logical_line.end(), m_matches_ll, dfh_lookup))
+        {
+            auto r = try_substituting_dfh(m_matches_ll);
+            if (r.error())
+            {
+                if (m_diags)
+                    m_diags->add_diagnostic(
+                        diagnostic_op::warn_CIC002(range(position(lineno, 0)), r.error_variable_name()));
+                m_pending_dfh_null_error = r.error_variable_name();
+            }
+            else if (r.substitutions_performed() > 0)
+                ret_val = true;
+        }
+
+        if (static const stmt_part_ids part_ids { 1, { 2 }, 3, 4 };
+            auto stmt = get_preproc_statement<semantics::cics_statement_si>(m_matches_ll, part_ids, lineno))
```

Comment:
`m_matches_ll` might be invalid when `m_logical_line.continuation_error` branch is taken?

Maybe the test in `get_preproc_statement` should be assert and return value should be guaranteed not to be nullptr?